lib/pages/media_view.dart

```diff
@@ -96,27 +108,64 @@ class MediaViewPage extends HookWidget {
               ],
             )
           : null,
-      body: GestureDetector(
-        onTapUp: (details) => showButtons.value = !showButtons.value,
-        onVerticalDragEnd: isZoomedOut.value
-            ? (details) {
-                if (details.primaryVelocity.abs() > 1000) {
+      body: Listener(
+        onPointerMove: isZoomedOut.value
+            ? (event) {
+                isDragging.value = true;
+                offset.value += event.delta;
+              }
+            : null,
+        onPointerCancel: (_) => offset.value = Offset.zero,
+        onPointerUp: isZoomedOut.value
+            ? (_) {
+                isDragging.value = false;
+                final speed = (offset.value - prevOffset).distance;
+                if (speed > speedThreshold ||
+                    offset.value.dy.abs() > yThreshold) {
                   Navigator.of(context).pop();
+                } else {
+                  offset.value = Offset.zero;
                 }
               }
             : null,
-        child: PhotoView(
-          scaleStateChangedCallback: (value) {
-            isZoomedOut.value = value == PhotoViewScaleState.zoomedOut ||
-                value == PhotoViewScaleState.initial;
-          },
-          minScale: PhotoViewComputedScale.contained,
-          initialScale: PhotoViewComputedScale.contained,
-          imageProvider: CachedNetworkImageProvider(url),
-          heroAttributes: PhotoViewHeroAttributes(tag: url),
-          loadingBuilder: (context, event) =>
-              const Center(child: CircularProgressIndicator()),
-        ),
+        child: Stack(children: [
+          AnimatedPositioned(
+            duration: isDragging.value
+                ? Duration.zero
+                : const Duration(milliseconds: 100),
+            top: offset.value.dy,
+            bottom: -offset.value.dy,
+            left: offset.value.dx,
+            right: -offset.value.dx,
+            child: AnimatedContainer(
+              transform: Matrix4Transform()
+                  .scale(max(0.9, 1 - offset.value.dy.abs() / 1000))
+                  .rotate(min(-offset.value.dx / 1000, 0.3))
+                  .matrix4,
+              duration: isDragging.value
+                  ? Duration.zero
+                  : const Duration(milliseconds: 100),
+              child: PhotoView(
+                backgroundDecoration:
+                    const BoxDecoration(color: Colors.transparent),
+                scaleStateChangedCallback: (value) {
+                  isZoomedOut.value = value == PhotoViewScaleState.zoomedOut ||
+                      value == PhotoViewScaleState.initial;
+                  showButtons.value = isZoomedOut.value;
+                },
+                onTapUp: (_, __, ___) {
+                  showButtons.value = !showButtons.value;
+                },
```

Comment:
After some testing this seems to cause a problem where the image gets stuck in a transformed position (`onPointerUp` never gets called)